packages/remix-dev/config/routes.ts

```diff
@@ -54,6 +54,13 @@ export interface DefineRouteOptions {
    * Should be `true` if this is an index route that does not allow child routes.
    */
   index?: boolean;
+
+  /**
+   * An optional unique id string for this route, that will be appended to the file
+   * name e.g: `<file-name>--<id>`. Use this if you need to aggregate two or more
+   * routes with the same route file.
+   */
+  id?: string;
```

Comment:
The appending nature of this, as currently implemented, seems misleading to me.  These parameters align with the underlying react-router `<Route>` params.  But in RR you would do:

```
<Route id="home" path="/home" ...>

// Later in a component
let matches = useMatches();
matches.find(m => m.route.id === "home");
```

But in this case, if a user sets `id="home"` in their remix route config, the eventual actual `id` will be `{filename}--home`.

If we're letting users define id's manually, maybe we should just use that `id` verbatim and error if they provide duplicates?

Reply:
Hi @brophdawg11 

I've used this `filename` pattern because today is the way that remix guarantee a _auto-unique-ID_ per route. So to keep with this pattern I just append the "custom ID" in the route.id that was generated previously.

But your question/point it's valid in IMHO, if we need to change this what is the best way to "verbatim and error if they provide duplicates" just throws a `new Error` inside `routes.ts` file?

Reply:
@jacob-ebey You on board with my my suggestion above?

Reply:
> Hi @brophdawg11
> 
> I've used this `filename` pattern because today is the way that remix guarantee a _auto-unique-ID_ per route. So to keep with this pattern I just append the "custom ID" in the route.id that was generated previously.
> 
> But your question/point it's valid in IMHO, if we need to change this what is the best way to "verbatim and error if they provide duplicates" just throws a `new Error` inside `routes.ts` file?

Yes, it should just throw an Error. We'll catch that elsewhere and surface the message in the CLI.